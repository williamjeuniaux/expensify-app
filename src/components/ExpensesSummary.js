import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => {
  return (
    <div>
    <p>Viewing {props.expenseCount}
      {props.expenseCount > 1 ? ' expenses ' : ' expense '}
      totalling {numeral(props.expensesTotal / 100).format('$0,0.00')}
    </p>
  </div>
)
}
const mapStateToProps = (state) => {
  const expenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: expenses.length,
    expensesTotal: selectExpensesTotal(expenses)
  }
}

export default connect(mapStateToProps)(ExpensesSummary);