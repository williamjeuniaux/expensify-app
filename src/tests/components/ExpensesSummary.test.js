import React from 'react';
import { shallow } from 'enzyme';
import {ExpensesSummary}w from '../../components/ExpensesSummary';

test('should summarize 1 expense properly', ()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={94.34}/>);
    expect(wrapper).toMatchSnapshot();
})

test('should summarize 2 expenses properly',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expenseTotal={150.34}/>);
    expect(wrapper).toMatchSnapshot();
})