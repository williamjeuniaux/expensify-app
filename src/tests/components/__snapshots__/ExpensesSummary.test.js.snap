// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should summarize 1 expense properly 1`] = `
<div>
  <p>
    Viewing 
    1
     expense 
    totalling 
    $0.00
  </p>
</div>
`;

exports[`should summarize 2 expenses properly 1`] = `
<div>
  <p>
    Viewing 
    2
     expenses 
    totalling 
    $0.00
  </p>
</div>
`;
